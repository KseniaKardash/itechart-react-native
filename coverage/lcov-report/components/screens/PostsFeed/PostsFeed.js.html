<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/screens/PostsFeed/PostsFeed.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">components/screens/PostsFeed</a> PostsFeed.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.65% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/17</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.65% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/17</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* @flow */
import React, { PureComponent } from "react";
import { View, StyleSheet, FlatList, TextInput } from "react-native";
import ButtonIcon from "../../common/ButtonIcon";
import HeaderTitle from "../../common/HeaderTitle";
import UserPost from "../../common/UserPost";
import type { Post, Posts } from "../../../types/types";
import FadeWrapper from "../../common/FadeWrapper";
import { SHADOW_COLOR } from "../../../constants/colors";
&nbsp;
type Props = {
  navigator: Object,
  searchName: string,
  toggleSearchStatus: boolean,
  setToggleSearchStatus: Function,
  filterPostsByUserName: Function,
  posts: Posts
};
&nbsp;
class PostsFeed extends PureComponent&lt;Props&gt; {
  onChangeText = <span class="fstat-no" title="function not covered" >(t</span>ext: string) =&gt; {
    const { filterPostsByUserName } = <span class="cstat-no" title="statement not covered" >this.props;</span>
<span class="cstat-no" title="statement not covered" >    filterPostsByUserName(text);</span>
  };
&nbsp;
  onShowSelectedPost = <span class="fstat-no" title="function not covered" >(i</span>d: string) =&gt; {
    const { posts, navigator } = <span class="cstat-no" title="statement not covered" >this.props;</span>
    const selectedPost = <span class="cstat-no" title="statement not covered" >posts.find(<span class="fstat-no" title="function not covered" >po</span>st =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      return post.id === id;</span>
    });
<span class="cstat-no" title="statement not covered" >    navigator.push({</span>
      screen: "Post",
      title: selectedPost.userName,
      passProps: {
        post: { ...selectedPost }
      },
      backButtonHidden: true,
      animated: true,
      animationType: "fade"
    });
  };
&nbsp;
  setToggle = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const { setToggleSearchStatus, toggleSearchStatus } = <span class="cstat-no" title="statement not covered" >this.props;</span>
<span class="cstat-no" title="statement not covered" >    setToggleSearchStatus(!toggleSearchStatus);</span>
  };
&nbsp;
  getKeyExtractor = <span class="fstat-no" title="function not covered" >(i</span>tem: Post) =&gt; <span class="cstat-no" title="statement not covered" >item.id;</span>
&nbsp;
  navigateToNextPage = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const { navigator, setToggleSearchStatus, toggleSearchStatus } = <span class="cstat-no" title="statement not covered" >this.props;</span>
<span class="cstat-no" title="statement not covered" >    navigator.push({</span>
      screen: "ImagePickerScreen",
      title: "ImagePickerScreen",
      backButtonHidden: true,
      animated: true,
      animationType: "fade"
    });
<span class="cstat-no" title="statement not covered" >    if (toggleSearchStatus) {</span>
<span class="cstat-no" title="statement not covered" >      setToggleSearchStatus(!toggleSearchStatus);</span>
    }
  };
&nbsp;
  renderItem = <span class="fstat-no" title="function not covered" >(i</span>nboundData: { item: Post }) =&gt; {
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;FadeWrapper&gt;
        &lt;UserPost
          userName={inboundData.item.userName}
          uri={{ uri: inboundData.item.uri }}
          uriPhoto={{ uri: inboundData.item.uriPhoto }}
          id={inboundData.item.id}
          onShowSelectedPost={this.onShowSelectedPost}
        /&gt;
      &lt;/FadeWrapper&gt;
    );
  };
&nbsp;
  render() {
    const { searchName, toggleSearchStatus, posts } = this.props;
    return (
      &lt;View style={styles.container}&gt;
        &lt;View style={styles.header}&gt;
          &lt;ButtonIcon iconName="search" onPress={this.setToggle} /&gt;
          &lt;HeaderTitle text="FEED" /&gt;
          &lt;ButtonIcon iconName="plus" onPress={this.navigateToNextPage} /&gt;
        &lt;/View&gt;
        {toggleSearchStatus ? (
<span class="branch-0 cbranch-no" title="branch not covered" >          &lt;TextInput</span>
            style={styles.textInput}
            placeholder="Search"
            value={searchName}
            onChangeText={this.onChangeText}
          /&gt;
        ) : (
          &lt;View /&gt;
        )}
        &lt;FlatList
          removeClippedSubviews={false}
          showsVerticalScrollIndicator={false}
          initialNumToRender={15}
          maxToRenderPerBatch={20}
          data={posts}
          windowSize={21}
          keyExtractor={this.getKeyExtractor}
          renderItem={this.renderItem}
          style={styles.flatList}
        /&gt;
      &lt;/View&gt;
    );
  }
}
&nbsp;
const styles = StyleSheet.create({
  container: {
    width: "100%",
    flex: 1,
    padding: 20,
    paddingTop: 10,
    backgroundColor: SHADOW_COLOR
  },
  header: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between"
  },
  textInput: {
    marginBottom: 10,
    fontSize: 18
  },
  flatList: {
    width: "100%"
  }
});
&nbsp;
export default PostsFeed;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Sep 26 2018 15:09:55 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
