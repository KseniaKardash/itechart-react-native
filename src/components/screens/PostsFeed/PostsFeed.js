/* @flow */
import React, { PureComponent } from "react";
import { View, StyleSheet, FlatList, TextInput } from "react-native";
import { connect } from "react-redux";
import ButtonIcon from "../../common/ButtonIcon";
import HeaderTitle from "../../common/HeaderTitle";
import UserPost from "../../common/UserPost";
import type { Post, Posts } from "../../../types/types";
import FadeWrapper from "../../common/FadeWrapper";
import { SHADOW_COLOR } from "../../../constants/colors";
import {
  changeSearchName,
  setToggleSearchStatus
} from "../../../actions/postsFeedAction";

type Props = {
  navigator: Object,
  searchName: string,
  toggleSearchStatus: boolean,
  changeSearchName: Function,
  setToggleSearchStatus: Function,
  posts: Posts
};

class PostsFeed extends PureComponent<Props> {
  onChangeText = (text: string) => {
    const { changeSearchName } = this.props;
    changeSearchName(text);
  };

  onShowSelectedPost = (id: string) => {
    const { posts } = this.props;
    const selectedPost = posts.find(post => {
      return post.id === id;
    });
    const { navigator } = this.props;
    navigator.push({
      screen: "Post",
      title: selectedPost.userName,
      passProps: {
        post: selectedPost
      },
      backButtonHidden: true,
      animated: true,
      animationType: "fade"
    });
  };

  setToggle = () => {
    const { setToggleSearchStatus, toggleSearchStatus } = this.props;
    setToggleSearchStatus(!toggleSearchStatus);
  };

  getKeyExtractor = (item: Post) => item.id;

  navigateToNextPage = () => {
    const { navigator } = this.props;
    navigator.push({
      screen: "ImagePickerScreen",
      title: "ImagePickerScreen",
      backButtonHidden: true,
      animated: true,
      animationType: "fade"
    });
  };

  filterPosts = (posts: Array<Post>, query: string): Array<Post> => {
    return posts.filter(post => {
      const userName = post.userName.toUpperCase();
      const queryData = query.toUpperCase();
      return userName.indexOf(queryData) !== -1;
    });
  };

  renderItem = (inboundData: { item: Post }) => {
    return (
      <FadeWrapper>
        <UserPost
          userName={inboundData.item.userName}
          uri={{ uri: inboundData.item.uri }}
          uriPhoto={{ uri: inboundData.item.uriPhoto }}
          id={inboundData.item.id}
          onShowSelectedPost={this.onShowSelectedPost}
        />
      </FadeWrapper>
    );
  };

  render() {
    const { searchName, toggleSearchStatus, posts } = this.props;
    let filteredPosts = this.filterPosts(posts, searchName);
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <ButtonIcon iconName="search" onPress={this.setToggle} />
          <HeaderTitle text="FEED" />
          <ButtonIcon iconName="plus" onPress={this.navigateToNextPage} />
        </View>
        {toggleSearchStatus ? (
          <TextInput
            style={styles.textInput}
            placeholder="Search"
            value={searchName}
            onChangeText={this.onChangeText}
          />
        ) : (
          <View />
        )}
        <FlatList
          removeClippedSubviews={false}
          showsVerticalScrollIndicator={false}
          initialNumToRender={15}
          maxToRenderPerBatch={20}
          data={filteredPosts}
          windowSize={21}
          keyExtractor={this.getKeyExtractor}
          renderItem={this.renderItem}
          style={styles.flatList}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
    flex: 1,
    padding: 20,
    paddingTop: 10,
    backgroundColor: SHADOW_COLOR
  },
  header: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between"
  },
  textInput: {
    marginBottom: 10,
    fontSize: 18
  },
  flatList: {
    width: "100%"
  }
});

const mapStateToProps = state => {
  return {
    searchName: state.postsFeed.searchName,
    toggleSearchStatus: state.postsFeed.toggleSearchStatus,
    posts: state.postsReducer.posts
  };
};

const mapDispatchToProps = dispatch => {
  return {
    changeSearchName: value => dispatch(changeSearchName(value)),
    setToggleSearchStatus: value => dispatch(setToggleSearchStatus(value))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PostsFeed);
