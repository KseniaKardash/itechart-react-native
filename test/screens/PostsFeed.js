import React from "react";
import renderer from "react-test-renderer";
import PostsFeed from "../../src/components/screens/PostsFeed/PostsFeed";

describe("PostsFeed snapShot", () => {
  it("renders as expected", () => {
    const snap = renderer.create(<PostsFeed />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
