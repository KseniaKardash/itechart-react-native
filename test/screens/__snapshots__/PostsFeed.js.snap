// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsFeed snapShot renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eaeaea",
      "flex": 1,
      "padding": 20,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#D14DAE",
              "fontSize": 40,
            },
            Object {
              "margin": 5,
              "marginBottom": 10,
              "marginTop": 10,
              "padding": 5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#545454",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": 20,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      FEED
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "#D14DAE",
              "fontSize": 40,
            },
            Object {
              "margin": 5,
              "marginBottom": 10,
              "marginTop": 10,
              "padding": 5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View />
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={15}
    keyExtractor={[Function]}
    maxToRenderPerBatch={20}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
